crates/extensions/c8y_config_manager/src/actor.rs

```diff
@@ -106,15 +105,18 @@ impl ConfigManagerActor {
                     if let Ok(config_download_request) = maybe_config_download_request {
                         if let Err(err) = self
                             .config_download_manager
-                            .handle_config_download_request(config_download_request, message_box)
+                            .handle_config_download_request(
+                                config_download_request,
+                                &mut self.messages,
+                            )
                             .await
                         {
                             Self::fail_config_operation_in_c8y(
                                 ConfigOperation::Update,
                                 None,
                                 ActiveOperationState::Pending,
                                 format!("Failed due to {}", err),
-                                message_box,
+                                &mut self.messages,
```

Comment:
Maybe , something can be improved here as promoting `fail_config_operation_in_c8y` as a method to avoid passing `self.messages`.

@albinsuresh what do you think about that?